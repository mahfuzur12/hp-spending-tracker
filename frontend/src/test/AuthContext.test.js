import React from "react";
import { render } from "@testing-library/react";
import { AuthContext, AuthContextProvider } from "../context/AuthContext.js";

describe("AuthContextProvider", () => {
  it("renders without crashing", () => {
    render(
      <AuthContextProvider>
        <div>Test</div>
      </AuthContextProvider>
    );
  });

  it("provides the initial state values to its children", () => {
    const { container } = render(
      <AuthContextProvider>
        <div>{({ user, isLoggedIn, token }) => `${user} ${isLoggedIn} ${token}`}</div>
      </AuthContextProvider>
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});