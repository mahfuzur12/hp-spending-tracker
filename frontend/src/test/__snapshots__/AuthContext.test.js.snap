// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthContextProvider provides the initial state values to its children 1`] = `<div />`;
